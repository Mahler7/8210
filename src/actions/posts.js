import uuid from 'uuid';

// export const addPost = (post) => ({
//   type: 'ADD_POST',
//   post
// });

export const addPost = (
  {
    title = '',
    content = '',
    createdAt = 0,
    author = ''
  } = {}
) => ({
  type: 'ADD_POST',
  post: {
    id: uuid(),
    title,
    content,
    createdAt,
    author
  }
});
console.log(addPost());