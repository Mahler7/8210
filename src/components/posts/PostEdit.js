import React from 'react';

const PostEdit = () => (
  <h1>Edit Post</h1>
);

export default PostEdit;