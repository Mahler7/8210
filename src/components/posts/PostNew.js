import React from 'react';

const PostNew = () => (
  <h1>New Post</h1>
);

export default PostNew;