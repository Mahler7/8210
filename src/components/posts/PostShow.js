import React from 'react';

const PostShow = () => (
  <h1>Show Post</h1>
);

export default PostShow;