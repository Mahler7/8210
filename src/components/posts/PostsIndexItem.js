import React from 'react';

const PostsIndexItem = ({ title, content, createdAt, author }) => (
  <div>
    <h1>{title}</h1>
    <span><i>{author}</i>-<i>{createdAt}</i></span>
    <p>{content}</p>
  </div>
);

export default PostsIndexItem;