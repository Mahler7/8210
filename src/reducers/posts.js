const postsReducerDefaultState = [];

const postsReducer = (state=postsReducerDefaultState, action) => {
  switch(action.type){
    case 'ADD_POST':
      return [
        ...state,
        action.post
      ]
    default: 
      return state
  }
};
  
export default postsReducer;