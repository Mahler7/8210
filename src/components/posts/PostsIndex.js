import React from 'react';
import { connect } from 'react-redux';
import PostsIndexItem from './PostsIndexItem';

export const PostsIndex = (props) => (
  <div>
    <h1>Posts Index</h1>
    {props.posts.map((post) => {
      return <PostsIndexItem key={post.id}{...post}/>
    })}
  </div>
);

const mapStateToProps = (state) => {
  return {
    posts: state.posts
  };
}

export default connect(mapStateToProps)(PostsIndex);