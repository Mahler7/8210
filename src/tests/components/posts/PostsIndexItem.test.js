import React from 'react';
import { shallow } from 'enzyme';
import posts from '../../fixtures/posts';
import PostIndexItem from '../../../components/posts/PostsIndexItem';

test('should render PostsIndex page', () => {
  const wrapper = shallow(<PostIndexItem {...posts[2]}/>);
  expect(wrapper).toMatchSnapshot();
});