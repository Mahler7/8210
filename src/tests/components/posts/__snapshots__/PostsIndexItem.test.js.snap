// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PostsIndex page 1`] = `
<div>
  <h1>
    Blog Post 1
  </h1>
  <span>
    <i>
      Dave
    </i>
    -
    <i>
      January 10th 1970
    </i>
  </span>
  <p>
    This is the third blog article
  </p>
</div>
`;
